routes/repository.js

```diff
@@ -0,0 +1,63 @@
+const express = require('express');
+const fs = require('fs');
+const path = require('path');
+const simpleGit = require('simple-git');
+const createError = require('http-errors');
+
+const router = express.Router();
+const repoUrlValidator = require('../middlewares/repoUrlValidator');
+const { changeBranchNameFormat, getRepoName } = require('../utils');
+const ERROR = require('../constants/errorConstants');
+const GIT = require('../constants/gitConstants');
+
+router.get('/', repoUrlValidator, async (req, res, next) => {
+  const logOption = [GIT.LOG_OPTION_ALL];
+  const cloneOption = [GIT.CLONE_OPTION_NO_CHECK_OUT];
+  const formatOptions = {
+    format: GIT.PRETTY_FORMAT_OPTIONS,
+  };
+
+  try {
+    const { repoUrl } = req.query;
+
+    const repoName = getRepoName(repoUrl);
+
+    try {
+      await simpleGit().clone(repoUrl, cloneOption);
+    } catch (err) {
+      throw createError(500, ERROR.FAIL_TO_CLONE);
+    }
+
+    const clonedGit = await simpleGit(path.resolve(`./${repoName}`));
+
+    if (!clonedGit) {
+      throw createError(400, ERROR.GIT_NOT_FOUND);
+    }
+
+    const log = await clonedGit.log(logOption, formatOptions);
+    const logList = log.all;
+
+    const formattedLogList = changeBranchNameFormat(logList);
+
+    if (!log) {
+      throw createError(401, ERROR.FAIL_TO_LOG);
+    }
```

Comment:
```suggestion
    const logList = await clonedGit.log(logOption, formatOptions).all;

    const formattedLogList = changeBranchNameFormat(logList);

    if (!logList ) {
      throw createError(401, ERROR.FAIL_TO_LOG);
    }
```

이런식으로도  쓸 수 있지안을까요??

Reply:
git.log 가 비동기 작업이라 resolve then 후에 all 값을 받는 것이 필요합니다.

Reply:
```js
const { all: logs } = await clonedGit.log(logOption, formatOptions);
```